File: solutions/ruby/darts/1/darts.rb
class Darts  
  def initialize(x, y)
    @dist = Math.sqrt(x*x+y*y)
  end

  def score
    return 10 if @dist <= 1
    return 5 if @dist <= 5
    return 1 if @dist <= 10
    return 0
  end

end

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

namespace darts {

unsigned int score(double x, double y) {
    unsigned int result = 0;
    auto distance_from_center = sqrt(x*x+y*y);
    if (distance_from_center <= 10) {
        result++;
        if (distance_from_center <= 5) {
            result += 4;        
            if (distance_from_center <= 1) {
                result += 5;        
            }
        }        
    }
    return result;
}

}  // namespace darts
